File: problems/2026-06-25/job-rate-limiter/runner.ts
(function _run() {
  const _rl = new RateLimiter(3, 60_000);
  const _t = 0;
  const _cases = [
    { id: 1, label: "1st request allowed",      call: () => _rl.isAllowed("alice", _t),           expected: true  },
    { id: 2, label: "2nd request allowed",      call: () => _rl.isAllowed("alice", _t + 1),       expected: true  },
    { id: 3, label: "3rd request allowed",      call: () => _rl.isAllowed("alice", _t + 2),       expected: true  },
    { id: 4, label: "4th request blocked",      call: () => _rl.isAllowed("alice", _t + 3),       expected: false },
    { id: 5, label: "different client allowed", call: () => _rl.isAllowed("bob",   _t + 3),       expected: true  },
    { id: 6, label: "after window, allowed",    call: () => _rl.isAllowed("alice", _t + 61_000),  expected: true  },
  ];
  const _results = _cases.map(c => {
    let _actual: unknown;
    try { _actual = c.call(); } catch (_e: any) { _actual = `ERROR: ${_e.message}`; }
    return { id: c.id, label: c.label, passed: JSON.stringify(_actual) === JSON.stringify(c.expected), expected: c.expected, actual: _actual };
  });
  const _passed = _results.filter(r => r.passed).length;
  process.stdout.write("__RESULT__:" + JSON.stringify({ results: _results, summary: { passed: _passed, total: _results.length } }) + "\n");
})();

File: problems/2026-06-25/job-rate-limiter/runner.py
import json as _json


def _run():
    _rl = RateLimiter(max_requests=3, window_ms=60_000)
    _t = 0
    _cases = [
        (1, "1st request allowed",      lambda: _rl.is_allowed("alice", _t),           True),
        (2, "2nd request allowed",      lambda: _rl.is_allowed("alice", _t + 1),       True),
        (3, "3rd request allowed",      lambda: _rl.is_allowed("alice", _t + 2),       True),
        (4, "4th request blocked",      lambda: _rl.is_allowed("alice", _t + 3),       False),
        (5, "different client allowed", lambda: _rl.is_allowed("bob",   _t + 3),       True),
        (6, "after window, allowed",    lambda: _rl.is_allowed("alice", _t + 61_000),  True),
    ]
    _results = []
    for _id, _label, _call, _expected in _cases:
        try:
            _actual = _call()
        except Exception as _e:
            _actual = f"ERROR: {_e}"
        _results.append({
            "id": _id, "label": _label,
            "passed": _actual == _expected,
            "expected": _expected, "actual": _actual,
        })
    _passed = sum(1 for r in _results if r["passed"])
    print("__RESULT__:" + _json.dumps({"results": _results, "summary": {"passed": _passed, "total": len(_results)}}))


_run()
